### ISSUE TITLE
Nested Class/Enum definitions fail

### ISSUE THREAD

**Issue by moosichu** [trung/InMemoryJavaCompiler#7]
Running the following:

``` java
sourceCode.append("package org.mdkt;\n");
sourceCode.append("public class HelloClass {\n");
sourceCode.append("   public static class Nested {}\n");
sourceCode.append("}");

System.out.println(sourceCode.toString());
Class<?> helloClass = InMemoryJavaCompiler.compile("org.mdkt.HelloClass",
        sourceCode.toString());`
```

Results in:

```
java.lang.NoClassDefFoundError: org/mdkt/HelloClass (wrong name: org/mdkt/HelloClass$Nested)
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
    at org.mdkt.compiler.DynamicClassLoader.findClass(DynamicClassLoader.java:28)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    at org.mdkt.compiler.InMemoryJavaCompiler.compile(InMemoryJavaCompiler.java:22)
    at uk.co.ensoft.xrm2m.codegenerator.SchemaClassGeneratorTest.testGenerateClass1(SchemaClassGeneratorTest.java:79)
```

**Comment by GreyCat** [comment]
@moosichu This fork seems to work: https://github.com/turpid-monkey/InMemoryJavaCompiler

**Comment by miho** [comment]
Actually, there's also a pull request of this fork: see pull request #2 : https://github.com/trung/InMemoryJavaCompiler/pull/2

**Comment by GreyCat** [comment]
Well, according to [this note](https://github.com/trung/InMemoryJavaCompiler/commit/8ef7356515cad21ea1e2fbd6880d07ade97d39c2), @trung might return and pull that stuff back into upstream git.

**Comment by miho** [comment]
cool.

**Comment by trung** [comment]
Thanks guys. I'm back. I have merged #2 and made some minor modification.
